### PR TITLE
And check conservation of water also without PrognosticSurfaceTemperature

### PR DESCRIPTION
**Pull request by Sbozzolo**
And check conservation of water also without PrognosticSurfaceTemperature

### CODE REVIEW

File: src/solver/solve.jl [hunk 1]
```diff
@@ -242,21 +242,19 @@ function check_conservation(sol)
         ) / energy_total
 
     mass_conservation =
-        (sum(sol.u[end].c.ρ) - sum(sol.u[1].c.ρ)) / sum(sol.u[1].c.ρ)
+        (sum(sol.u[end].c.ρ) - sum(sol.u[1].c.ρ)) / sum(sol.u[begin].c.ρ)
```
Comment (szy21) [inline src/solver/solve.jl:245]:
```suggestion
        (sum(sol.u[end].c.ρ) - sum(sol.u[begin].c.ρ)) / sum(sol.u[begin].c.ρ)
```
Also @dennisYatunin prefers 1 over begin. I don't have a preference as long as we are consistent:)

File: src/solver/solve.jl [hunk 2]
```diff
@@ -242,21 +242,19 @@ function check_conservation(sol)
...
 
-    water_conservation = zero(Spaces.undertype(axes(sol.u[end].c.ρ)))
+    water_surface_change = zero(Spaces.undertype(axes(sol.u[end].c.ρ)))
+    water_total = sum(sol.u[begin].c.ρq_tot)
+    water_atmos_change = sum(sol.u[end].c.ρq_tot) - sum(sol.u[1].c.ρq_tot)
```
Comment (szy21) [inline src/solver/solve.jl:249]:
```suggestion
    water_atmos_change = sum(sol.u[end].c.ρq_tot) - sum(sol.u[begin].c.ρq_tot)
```

**Comment by szy21** [comment]
The issue with using t=0 is that `water_total` will be zero for most of the cases (where the initial condition is dry).

**Comment by Sbozzolo** [comment]
> The issue with using t=0 is that `water_total` will be zero for most of the cases (where the initial condition is dry).

Okay, it makes sense.

**Comment by Sbozzolo** [comment]
> The issue with using t=0 is that `water_total` will be zero for most of the cases (where the initial condition is dry).

I rolled back to checking with `end` (even if I think we should not do that), and singled out the change to check water conservation also without PrognosticSurfaceTemperature.

**Review by szy21 — APPROVED** [review]
Looks good. Just keep in mind that when sfc is not Prognostic, water conservation will not hold when there is source or sink of moisture (evaporation or precipitation).

File: src/solver/solve.jl [hunk 3]
```diff
@@ -241,26 +241,20 @@ function check_conservation(sol)
             energy_radiation_input,
         ) / energy_total
 
-    mass_conservation =
-        (sum(sol.u[end].c.ρ) - sum(sol.u[1].c.ρ)) / sum(sol.u[1].c.ρ)
-
-    water_conservation = zero(Spaces.undertype(axes(sol.u[end].c.ρ)))
+    water_surface_change = zero(Spaces.undertype(axes(sol.u[end].c.ρ)))
+    water_total = sum(sol.u[1].c.ρq_tot)
```
Comment (szy21) [inline src/solver/solve.jl:245]:
```suggestion
    water_total = sum(sol.u[end].c.ρq_tot)
```
This should fix the error here: https://buildkite.com/clima/climaatmos-ci/builds/19579#01907544-653d-482c-ae12-4c964b6778c9/200-485

**Comment by szy21** [comment]
Actually I want to output water conservation check without PrognosticSurfaceTemperature as well. I'll push to this PR.

**Pull request merged**